src/Interval.cpp

```diff
@@ -157,5 +157,115 @@ Expr Interval::neg_inf_noinline() {
     return Interval::neg_inf_expr;
 }
 
+ConstantInterval::ConstantInterval() = default;
+
+ConstantInterval::ConstantInterval(int64_t min, int64_t max)
+    : min(min), max(max), min_defined(true), max_defined(true) {
+}
+
+ConstantInterval ConstantInterval::everything() {
+    return ConstantInterval();
+}
+
+ConstantInterval ConstantInterval::nothing() {
```

Comment:
Hmm, I've left a couple of comments talking about issues with this, but I don't think I actually see this function called/used anywhere. Probably is worth removing (along with `ConstantInterval::is_empty()`)

Reply:
This is what I ended up doing, it avoids a lot of the other problems you pointed out.